applications/20acts/contracts/src/TwentyActs.sol

```diff
@@ -0,0 +1,810 @@
+    uint256 public withdrawalWaitPeriod;
+
+    address public infrastructureAccount;
+
+    /**
+     *
+     * @param _withdrawalWaitPeriod Period between requesting a withdraw and being able to action the withdraw in seconds.
+     */
+    constructor(
+        uint256 _myBlockchainId,
+        uint256 _withdrawalWaitPeriod,
+        address _infrastructureAccount
+    ) {
+        address sender = msg.sender;
+        _setupRole(DEFAULT_ADMIN_ROLE, sender);
+
+        _setupRole(MAPPING_ROLE, sender);
+        _setupRole(PAUSER_ROLE, sender);
+
+        myBlockchainId = _myBlockchainId;
+        withdrawalWaitPeriod = _withdrawalWaitPeriod;
+        infrastructureAccount = _infrastructureAccount;
+    }
+
+    /**
+     * @dev Set the period of time in seconds that an entity must wait prior to withdrawing funds.
+     * NOTE: Caller must have DEFAULT_ADMIN_ROLE
+     *
+     * @param _withdrawalWaitPeriod Period between requesting a withdraw and being able to action the withdraw in seconds.
+     */
+    function setWithdrawalWaitPeriod(uint256 _withdrawalWaitPeriod) external {
+        require(
+            hasRole(DEFAULT_ADMIN_ROLE, _msgSender()),
+            "20ACTS:Must have ADMIN role"
+        );
+
+        withdrawalWaitPeriod = _withdrawalWaitPeriod;
+        emit SetWithdrawalWaitPeriod(_withdrawalWaitPeriod);
+    }
+
+    /**
+     * @dev Pauses the bridge.
+     * NOTE: caller must have the PAUSER_ROLE.
+     */
+    function pause() external {
+        require(
+            hasRole(PAUSER_ROLE, _msgSender()),
+            "20ACTS:Must have PAUSER role"
+        );
+        _pause();
+    }
+
+    /**
+     * @dev Unpauses the bridge.
+     * NOTE: caller must have the PAUSER_ROLE.
+     */
+    function unpause() external {
+        require(
+            hasRole(PAUSER_ROLE, _msgSender()),
+            "20ACTS:Must have PAUSER role"
+        );
+        _unpause();
+    }
+
+    /**
+     * @dev Set or update the mapping between an ERC 20 contract on this blockchain and an ERC 20
+     * contract on another blockchain.
+     * NOTE: caller must have the MAPPING_ROLE.
+     *
+     * @param _thisBcErc20          Address of ERC 20 contract on this blockchain.
+     * @param _otherBcId            Blockchain ID where the corresponding ERC 20 contract resides.
+     * @param _otherBcErc20         Address of ERC 20 contract on the other blockchain. Set to 0 to
+     *                              remove the mapping, thus indicating that the token can not be
+     *                              transferred to blockchain _otherBcId.
+     */
+    function setErc20Mapping(
+        address _thisBcErc20,
+        uint256 _otherBcId,
+        address _otherBcErc20
+    ) external {
+        require(
+            hasRole(MAPPING_ROLE, _msgSender()),
+            "20ACTS:Must have MAPPING role"
+        );
+
+        // Indicate the token us supported.
+        erc20Supported[_thisBcErc20] = true;
```

Comment:
should we support a way to remove a ERC20 support from the contract?

Reply:
I have revised the documentation.
What ERC20 support means is that we support the ability for liquidity providers to deposit the tokens. As such, it is not a security issue, but more of a protection for the liquidity providers that they don't mistakenly add liquidity for a token we don't support. 
If the operators of the contract remove all of the mappings for an ERC 20 token, the deposited tokens won't be able to be transferred anywhere.

Reply:
I worked out a zero cost way of fixing this, and hence it has not been resolved. Rather than storing a boolean, have a counter.